numba/core/typing/context.py

```diff
@@ -385,6 +385,15 @@ def resolve_value_type(self, val):
 
         raise typeof_exc
 
+    def resolve_value_type_as_literal(self, value):
```

Comment:
```suggestion
    def resolve_value_type_prefer_literal(self, value):
```
I'm wondering if this ought to be renamed as this, as the type will still resolve if possible, it's just that literals are preferred opposed to required.